internal/oci/runtime_oci.go

```diff
@@ -521,6 +544,42 @@ func (r *runtimeOCI) ExecSyncContainer(ctx context.Context, c *Container, comman
 
 	// We don't need childPipe on the parent side
 	childPipe.Close()
+	childStartPipe.Close()
+
+	if r.handler.MonitorExecCgroup && r.config.InfraCtrCPUSet != "" {
```

Comment:
we need to defer the container process kill if it errors here (with a closure like in CreateContainer), otherwise we'll leak a process because conmon will be waiting forever on the pipe

Reply:
thanks for the comment... I've updated the code to do the same kill as CreateContainer